### ISSUE TITLE
[Bug] Sometime decode meets cache_miss

### ISSUE THREAD

**Issue by aolemila** [sgl-project/sglang-jax#81]
### Describe the bug

Sometime decode meets cache_miss.

<img width="1115" height="313" alt="Image" src="https://github.com/user-attachments/assets/91ff602e-8a43-4b6a-a973-462b9f0b65b3" />

### Reproduction

**Command**

`JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen-7B-Chat --trust-remote-code  --dist-init-addr=0.0.0.0:10011 --nnodes=1  --tp-size=4 --device=tpu --random-seed=3 --node-rank=0 --mem-fraction-static=0.4 --chunked-prefill-size=8192 --download-dir=/tmp --dtype=bfloat16 --jax-precompile-decode-bs-paddings 1 64 256 --max-running-requests 256 --skip-server-warmup --attention-backend=fa --jax-precompile-prefill-token-paddings 8192 --page-size=64`

**Trigger**

note: `evalscope<1.0.0`

`evalscope eval  --model Qwen-7B-Chat --api-url http://127.0.0.1:30000/v1/chat/completions --api-key EMPTY --eval-type service --datasets gsm8k --eval-batch-size 64`

### Environment

main-9e3c57e84dd8cf44eeb0f05ac4c51c943ad043cf

#### Accuracy

```bash
+--------------+-----------+-----------------+----------+-------+---------+---------+
| Model        | Dataset   | Metric          | Subset   |   Num |   Score | Cat.0   |
+==============+===========+=================+==========+=======+=========+=========+
| Qwen-7B-Chat | gsm8k     | AverageAccuracy | main     |  1319 |  0.4761 | default |
+--------------+-----------+-----------------+----------+-------+---------+---------+

```
